### PR TITLE
Do not alert lack of maintenance on inactive db

### LINKED ISSUE #129: last_vacuum should not return critical if there was no write since last vacuum
Hi,

When there is no write on a database since the last vacuum, vacuum does not run. Even a manual vacuum full analyse does not run the vacuum. So on a database where there is no write, the last_vacuum probe will always return CRITICAL. Would it be possible to detect this case?

http://grokbase.com/t/postgresql/pgsql-bugs/10asqyzkn2/bug-5722-vacuum-full-does-not-update-last-vacuum-statistics/oldest

Alternatively, it should be possible to run last_vacuum and last_analyse without thresholds, for graphing purposes only.

Thanks.

### PR DESCRIPTION
**Pull request by ioguix**
closes #129

### CODE REVIEW

**Comment by ioguix** [comment]
Hi,

The only question on this PR is what to do with the perfdata when the check returns OK despite the last maintenance is older than the given threshold. From the graphing UI, this could lead to metrics entering the read area without alerts.

Ideas?

Note that this could be solved later to avoid postponing last release infinitely.

**Comment by Krysztophe** [comment]
Sounds good to me, except the error messages when the status file is not initialized.

I've tested with `-w 1m` and the service does not raise  WARNING as long as nothing is written.
A simple INSERT,UPDATE,CREATE EXTENSION... raises the WARNING.

**Comment by Krysztophe** [comment]
Logically, the durations can attain extremely high values (months?).  I don't see it as a problem. 

Users may be astonished to see a WARNING suddenly raised with a very value, just after some administrative command or a very small change, asking why it wasn't raised before. There should be an hint in the doc.

File: check_pgactivity [hunk 1]
```diff
@@ -4701,10 +4723,10 @@ LOOP_DB: foreach my $db (@all_db) {
...
 
-        if ($hosts[0]->{'version_num'} >= $PG_VERSION_91 ) {
-            $new_counts{$db} = [$rs->[1], $rs->[2]];
+        $new_counts{$db} = [ $rs->[1], $rs->[2], $counts{$db}[2] ];
 
-            if ( defined $counts{$db} ) {
+        if ( defined $counts{$db} ) {
+            if ($hosts[0]->{'version_num'} >= $PG_VERSION_91 ) {
                 my $delta      = $rs->[1] - $counts{$db}[0];
                 my $delta_auto = $rs->[2] - $counts{$db}[1];
```
[the remark this answers is not in the document]
Comment (ioguix) [reply check_pgactivity:4742]: fixed! Thanks for the review!

**Comment by ioguix** [comment]
Hi @Krysztophe,

Thank you for your review.

I fixed the error you found, added some code comments and user documentation.

Let me know if it's OK for you.

Regards,

**Comment by Krysztophe** [comment]
Tested again on 8.2, 8.4, 9.1, this seems ok.
As said somewhere else: in 8.2, pg_stat_user_tables depends on `stats_*` parameters that are off by default, so I'm afraid that the service could decide wrongly that nothing happens in the DB if it just looks at this view.

**Comment by ioguix** [comment]
> in 8.2, pg_stat_user_tables depends on stats_* parameters that are off by default

This is a good catch. We should probably focus more often on services compatibility with current GUC values...Not just for very old release.

I introduced `is_guc()` to check this. Hopefully it will be used frequently in future releases.

**Comment by Krysztophe** [comment]
Looks good to me!

**Pull request merged**